File: src/main/java/com/example/the_machine/config/BootstrapConfig.java
package com.example.the_machine.config;

import com.example.the_machine.db.AssistantEntity;
import com.example.the_machine.repo.AssistantRepository;
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.boot.context.event.ApplicationReadyEvent;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.event.EventListener;
import org.springframework.transaction.annotation.Transactional;

@Configuration
@RequiredArgsConstructor
@Slf4j
public class BootstrapConfig {

  private final AssistantRepository assistantRepository;
  private final ObjectMapper objectMapper;

  @EventListener(ApplicationReadyEvent.class)
  @Transactional
  public void seedDefaultAssistant() {
    if (assistantRepository.count() == 0) {
      log.info("No assistants found, creating default assistant");

      try {
        JsonNode paramsJson = objectMapper.readTree("""
            {
              "temperature": 0.4,
              "maxOutputTokens": 2048
            }
            """);

        var defaultAssistant = AssistantEntity.builder()
            .name("Default Assistant")
            .model("anthropic/claude-3.7")
            .paramsJson(paramsJson)
            .build();

        assistantRepository.save(defaultAssistant);
        log.info("Created default assistant with ID: {}", defaultAssistant.getId());
      } catch (Exception e) {
        log.error("Failed to create default assistant", e);
        throw new RuntimeException("Failed to bootstrap default assistant", e);
      }
    } else {
      log.debug("Assistants already exist, skipping default assistant creation");
    }
  }
}

File: src/main/kotlin/com/example/the_machine/config/BootstrapConfig.kt
package com.example.the_machine.config

import com.example.the_machine.common.KotlinSerializationConfig
import com.example.the_machine.domain.AssistantEntity
import com.example.the_machine.repo.AssistantRepository
import mu.KotlinLogging
import org.springframework.boot.context.event.ApplicationReadyEvent
import org.springframework.context.annotation.Configuration
import org.springframework.context.event.EventListener
import org.springframework.transaction.annotation.Transactional

@Configuration
class BootstrapConfig(
  private val assistantRepository: AssistantRepository
) {

  private val log = KotlinLogging.logger {}

  @EventListener(ApplicationReadyEvent::class)
  @Transactional
  fun seedDefaultAssistant() {
    if (assistantRepository.count() == 0L) {
      log.info { "No assistants found, creating default assistant" }

      try {
        val jsonString = """
                    {
                      "temperature": 0.4,
                      "maxOutputTokens": 2048
                    }
                    """.trimIndent()

        val defaultAssistant = AssistantEntity(
          name = "Default Assistant",
          model = "anthropic/claude-3.7",
          paramsJson = KotlinSerializationConfig.staticJson.parseToJsonElement(jsonString)
        )

        val savedAssistant = assistantRepository.save<AssistantEntity>(defaultAssistant)
        log.info { "Created default assistant with ID: ${savedAssistant.id}" }
      } catch (e: Exception) {
        log.error(e) { "Failed to create default assistant" }
        throw RuntimeException("Failed to bootstrap default assistant", e)
      }
    } else {
      log.debug { "Assistants already exist, skipping default assistant creation" }
    }
  }
}